mongo/src/main/java/com/redhat/lightblue/mongo/crud/MongoCRUDController.java

```diff
@@ -658,8 +668,13 @@ private void createUpdateEntityInfoIndexes(EntityInfo ei) {
             for(Index index:createIndexes) {
                 LOGGER.info("Creating index {} with {}",index.getName(),index.getFields());
                 DBObject newIndex = new BasicDBObject();
-                for (SortKey p : index.getFields()) {
-                    newIndex.put(p.getField().toString(), p.isDesc() ? -1 : 1);
+                for (IndexSortKey p : index.getFields()) {
+                    String field = p.getField().toString();
+                    if (p.isCaseInsensitive()) {
+                        // if this is a case insensitive key, we need to change the field to how mongo actually stores the index
+                        field = Translator.getHiddenForField(p.getField()).toString();
+                    }
```

Comment:
p.getField().toString() is not correct. Instead, change getHiddenForfield accept a Path, not String, and use Translator.translatePath() to translate path.